Src/Base/AMReX_FabArrayBase.cpp

```diff
@@ -740,7 +757,19 @@ FabArrayBase::FB::define_fb(const FabArrayBase& fa)
 		const Box& dst_bx   = isects[j].second - *pit;
 		const int src_owner = dm[ksnd];
 		
-		const BoxList& bl = amrex::boxDiff(dst_bx, vbx);
+		BoxList bl = amrex::boxDiff(dst_bx, vbx);
+		
+        if (m_multi_ghost) 
+        {
+            // In the case where ngrow>1, augment the send/rcv box list
+            // with boxes for overlapping ghost nodes.
+            Box ba_ksnd = ba[ksnd];
+            ba_ksnd.grow(1);
+            const Box dst_bx_ng = (ba_ksnd & (bxrcv_ng + (*pit))) - (*pit);
+		    const BoxList& bl_ng = amrex::boxDiff(dst_bx_ng, vbx_ng);
+            bl.join(ba.complementIn(bl_ng));
```

Comment:
I don't think this works for periodic boundary.  The points in `bl_ng` might overlap with valid points periodically.  But if you don't have periodic boundaries.  We could add an assertion and have this PR merged.

Reply:
Also, thinking more about the approach in this PR, I don't think it will scale well because of the line above even without periodic boundaries.  I believe it will be slower than doing two communication passes unless the number of component is huge.  The two passes can be done as follows.  First, a modified FillBoundary is called to fill the second layer (or beyond) of ghost cells with the valid cells and the first layer of ghost cells.  Then a normal FillBoundary is called.  In the first pass, one can use this BoxArray member function for intersections
```
    void intersections (const Box& bx, std::vector< std::pair<int,Box> >& isects,
                        bool first_only, const IntVect& ng) const;
```
Here the effect of `ng` is to effectively grow BoxArray by `ng`.  Using this, we can treat the first layer of ghost cells as valid cells.

Reply:
In the first pass, the communication can be restricted to sending from first layer of ghost cells to the outer layers.

Reply:
> We could add an assertion and have this PR merged.

Assertion added.

>  I don't think it will scale well because of the line above even without periodic boundaries

Because of the `ba.complementIn(bl_ng)` call? 

> First, a modified FillBoundary is called to fill the second layer (or beyond) of ghost cells with the valid cells and the first layer of ghost cells. Then a normal FillBoundary is called. In the first pass, one can use this BoxArray member function for intersections

Ok, so in practice, one would pass an argument (say, `ng`) to `FillBoundary`, so that the total call would look like

    FillBoundary(ng) // modified fill boundary
    FillBoundary() // regular fill boundary

The `ng` argument for the modified `FillBoundary` call makes its way to `define_fb` where it is used in the `intersections` call. Is that you had in mind? If so, I'm happy to give it a try but I'm doubtful that it will work - I found that the bad `FillBoundary` behavior happened even when the nodes in question were already filled with the proper values.